doc/CONTRIBUTING.md

```diff
@@ -72,26 +72,46 @@ If you are not sure of what to do, don't hesitate to open an issue about your PR
 
 #### The commit message
 
-The commit message has to be clear and related to the PR content.
+Sonata is a big project with many contributors, and a big part of the job is
+being able to understand the code at all times, be it when submitting a PR or
+looking at the history. Good commit messages are crucial to achieve this goal.
 
-The first line of the commit message must be short.
-The other lines must contains a complete description of what you done and why.
+There are already a few articles (or even single purpose websites) about this,
+we cannot recommend enough the following:
 
-The description is optional but recommended. It could be asked by the team if needed.
+* http://rakeroutes.com/blog/deliberate-git
+* http://stopwritingramblingcommitmessages.com
+* http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
 
-Bad commit message:
+To sum them up, the commit message has to be crystal clear and of course,
+related to the PR content.
+
+The first line of the commit message must be short, keep it under 50
+characters. It must say concisely but *precisely* what you did. The other
+lines, if needed, should contain a complete description of *why* you did this.
+
+Bad commit message subject:
 
 ```
-Update Admin.php
+Update README.md
 ```
 
-Good commit message:
+Good commit message subject :
 
 ```
-Improve search indexing speed for sub-categories
+Document how to install the project
 ```
 
-Good commit message with description
+The description is optional but strongly recommended. It could be asked by the
+team if needed. PR will often lead to complicated, hard-to-read conversations
+with many links to other web pages.
+
+he commit description should be able to live without what is said in the PR,
+and should ideally sum it up in a crystal clear way, so that people do not have
+to open a web browser to understand what you did. Links to PR and external
```

Comment:
You're missing this one:

* New line on "Links to PR..."
* `to PR` => `to PRs/Issues`